### ISSUE TITLE
crash: spv::Builder::accessChainLoad(spv::Decoration, spv::Decoration, unsigned int, spv::MemoryAccessMask, spv::Scope, unsigned int)

### ISSUE THREAD

**Issue by paulthomson** [KhronosGroup/glslang#1821]
[bug_report.zip](https://github.com/KhronosGroup/glslang/files/3355474/bug_report.zip)

Issue found using [GraphicsFuzz](https://github.com/google/graphicsfuzz).

Tool versions:

* glslangValidator commit hash: 40c16ec0b3ad03fc170f1369a58e7bbe662d82cd

To reproduce:

`glslangValidator -V shader.comp -o shader.comp.spv`

The following shader files are included in the attached archive, some of which are also shown inline below:

* 0_glsl/shader.comp

0_glsl/shader.comp:

```
#version 310 es

layout(std430, binding = 0) buffer doesNotMatter {
 int result;
 int data[];
} ;
layout(local_size_x = 16, local_size_y = 1) in;
void main()
{
 float donor_replacementGLF_dead3ref;
 donor_replacementGLF_dead3ref = fwidth(74.05);
}

```

**Comment by paulthomson** [comment]
Ah I see that we are using fwidth in a compute shader. The debug build of glslang fails earlier with:

```
glslangValidator: ../glslang/MachineIndependent/ParseHelper.cpp:2212: void glslang::TParseContext::nonOpBuiltInCheck(const glslang::TSourceLoc&, const glslang::TFunction&, glslang::TIntermAggregate&): Assertion `PureOperatorBuiltins == false' failed.
```

**Comment by johnkslang** [comment]
So, we need to remove fwidth() from the compute shader, or make sure it is protected by the right extensions.

**Comment by arcady-lunarg** [comment]
This was fixed by #1837. Closing.
